libbeat/outputs/elasticsearch/client.go

```diff
@@ -178,6 +177,25 @@ func eventBulkMeta(index string, event common.MapStr) bulkMeta {
 	return meta
 }
 
+// getIndex returns the full index name
+// Index is either defined in the config as part of the output
+// or can be overload by the event through setting index
+func getIndex(event common.MapStr, index string) string {
+
+	ts := time.Time(event["@timestamp"].(common.Time)).UTC()
+
+	// Check if index is ste dynamically
+	if dynamicIndex, ok := event["index"]; ok {
```

Comment:
I wonder if index is the right key to use here, since I imagine it might show up as a key in some of the Beats. `@index` would be safer, IMHO.

Or we could keep index but have a way of disabling the dynamic index part. Actually, this should probably be "opt in", otherwise we risk someone adding an index key and suddenly all their data gets spread out.